### PR TITLE
Use redux in react components

### PR DESCRIPTION
**Pull request by techEmprez**
The following took place:
make sure that you keep state immutable in reducers!
change the default state in your books reducer from an empty array to an array with a few books.
Import Redux Provider and your store in the main component of your React app.
Connect app to the Redux store.
Make sure that list of books displays books from Redux store.
Use useSelector to consume the state.
add functionality for "Add new" button click:
Import Redux useDispatch.
Set the data inputs in the local React state (set title and author
Takes the inputs from the state and generate a unique id and add them to an object.
dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for removing books, add a functionality for "Remove" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator).
In your React component responsible for categories, add a functionality for "Check status" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator).

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,8 +5,9 @@ import {
   Route,
 } from 'react-router-dom';
 import './App.css';
+// eslint-disable-next-line import/extensions
+import Books from './components/addBook';
```
Comment (zainsadaqat) [inline src/App.js]: When I run your project, It shows me the following error:

Screenshot 2022-09-22 at 8 04 50 AM

As there's no file named `addBook` that's why It's throwing an error. The actual file name is `Addbook.js`. Kindly fix this 👍🏻

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
# Hello @techEmprez  😄 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Github Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ import {
   Route,
 } from 'react-router-dom';
 import './App.css';
+import Books from './components/Addbook';
 import NavBar from './components/navigation';
-import Books from './components/appBook';
```
Comment (BenMukebo) [inline src/App.js:9]: - [ ] Great job so far 👏🏻  However, the `pageCart` component responsible for rendering categories should contain a button for checking the status. This button would use the function `check status` to dispatches the `underConstruction` action creator (you can use a different function name). Kindly add this 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @techEmprez, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- List of books displays books from Redux store 👍🏻 
- Adding and removing books is possible. keep rocking :rocket: 
 ## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 2 above)
Comment (techEmprez) [reply src/App.js:9]: Done

File: src/App.js (hunk 1 above)
Comment (techEmprez) [reply src/App.js]: Done

**Review by brytebee — APPROVED** [review]
Hi, @techEmprez👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**